plugins/GSdx/GSSettingsDlg.cpp

```diff
@@ -712,7 +711,16 @@ void GSHacksDlg::OnInit()
 	CheckDlgButton(m_hWnd, IDC_FAST_TC_INV, theApp.GetConfigB("UserHacks_DisablePartialInvalidation"));
 	CheckDlgButton(m_hWnd, IDC_AUTO_FLUSH, theApp.GetConfigB("UserHacks_AutoFlush"));
 	CheckDlgButton(m_hWnd, IDC_UNSCALE_POINT_LINE, theApp.GetConfigB("UserHacks_unscale_point_line"));
-
+	if(ogl && theApp.m_gs_offset_hack.size()==2)
+	{
+		theApp.m_gs_offset_hack.push_back(GSSetting(2,  "Special (Texture)", ""));
+		theApp.m_gs_offset_hack.push_back(GSSetting(3,  "Special (Texture - aggressive)", ""));
+	}
+	else if (!ogl && theApp.m_gs_offset_hack.size()==4)
+	{
+		theApp.m_gs_offset_hack.erase(theApp.m_gs_offset_hack.begin() + 2,theApp.m_gs_offset_hack.begin() + 4); //We need to remove element 4 because function deletes elements between first and second including first but excluding second.
+	}
+	ComboBoxInit(IDC_OFFSETHACK, theApp.m_gs_offset_hack, std::min((theApp.m_gs_offset_hack.size()-1),(theApp.GetConfigI("UserHacks_HalfPixelOffset"))));
```

Comment:
I suggest going with something like this,

```
if(!ogl)
	{
		std::vector<GSSetting>::iterator head = theApp.m_gs_offset_hack.begin();
		std::vector<GSSetting> options(head, head + 2);
		const int combobox_limit = options.size() - 1;
		ComboBoxInit(IDC_OFFSETHACK, options, std::min(combobox_limit, theApp.GetConfigI("UserHacks_HalfPixelOffset")));
	}
	else
	{
		ComboBoxInit(IDC_OFFSETHACK, theApp.m_gs_offset_hack, theApp.GetConfigI("UserHacks_HalfPixelOffset"));
	}
```

It looks more readable in my humble opinion.